src/NHSD.BuyingCatalogue.Organisations.Api/Services/MailKitEmailService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using MailKit;
+using NHSD.BuyingCatalogue.Organisations.Api.Settings;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    /// <summary>
+    /// A service for sending e-mails using MailKit.
+    /// </summary>
+    internal sealed class MailKitEmailService : IEmailService
+    {
+        private readonly IMailTransport _client;
+        private readonly SmtpSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MailKitEmailService"/> class
+        /// using the provided <paramref name="client"/> and <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="client">The mail transport to use to send e-mail.</param>
+        /// <param name="settings">The SMTP configuration.</param>
+        public MailKitEmailService(IMailTransport client, SmtpSettings settings)
+        {
+            _client = client;
+
+            // TODO: confirm this is acceptable for us if using SSL
+            // (as we will be using a trusted server that is not
+            // user-configurable this should be fine, providing the
+            // root CA is trusted by the host system)
+            client.ServerCertificateValidationCallback = (s, c, h, e) => true;
+
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// Sends an e-mail message asynchronously using the
+        /// provided <see cref="IMailTransport"/> instance.
+        /// </summary>
+        /// <param name="emailMessage">The e-mail message to send asynchronously.</param>
+        /// <returns>An asynchronous task context.</returns>
+        public async Task SendEmailAsync(EmailMessage emailMessage)
```

Comment:
Could or should the email address parameter be null?

Reply:
It shouldn't be null but it could null although only through developer error at this point in time.

As I've commented above, while I'm in favour of failing fast I'm not always convinced by the value of precondition assertions in non-public code although I guess you could argue that's slightly contradictory... However, as both you and @CallumBradburyBJSS  have raised similar points I'm going to add them.

Possibly another one for code standards – do we want to do this everywhere? I guess nullable reference types might help here too (if we enabled that features).